aea/cli/generate.py

```diff
@@ -0,0 +1,124 @@
+import yaml
+
+# from aea import AEA_DIR
+from aea.cli.common import Context, pass_ctx, logger, _try_to_load_agent_config
+from aea.configurations.base import PublicId, DEFAULT_AEA_CONFIG_FILE
+from aea.protocols.generator import ProtocolTemplate, ProtocolGenerator, ProtocolSpecificationParseError
+
+# these variables are being used dynamically
+from aea.configurations.base import DEFAULT_CONNECTION_CONFIG_FILE, DEFAULT_PROTOCOL_CONFIG_FILE, DEFAULT_SKILL_CONFIG_FILE  # noqa: F401
+
+
+@click.group()
+@pass_ctx
+def generate(ctx: Context):
+    """Generate a resource for the agent."""
+    _try_to_load_agent_config(ctx)
+
+
+def _generate_item(ctx: Context, item_type, item_name, specification_path):
+    """Generate an item based on a specification and add it to the configuration file and agent."""
+    existing_id_list = getattr(ctx.agent_config, "{}s".format(item_type))
+    existing_item_list = [public_id.name for public_id in existing_id_list]
+
+    # loader = getattr(ctx, "{}_loader".format(item_type))
+    # default_config_filename = globals()["DEFAULT_{}_CONFIG_FILE".format(item_type.upper())]
+
+    item_type_plural = item_type + "s"
+
+    protocol_template = ProtocolTemplate(specification_path)
+
+    try:
+        protocol_template.load()
+    except (yaml.YAMLError, ProtocolSpecificationParseError) as e:
+        print(str(e))
+        print("Load error, exiting now!")
+        exit(1)
+
+    # check if we already have an item with the same name
+    logger.debug("{} already supported by the agent: {}".format(item_type_plural, existing_item_list))
+    protocol_name = protocol_template.name
+    if protocol_name in existing_item_list:
+        logger.error("A {} with name '{}' already exists. Aborting...".format(item_type, protocol_name))
+        sys.exit(1)
+
+    # import pdb; pdb.set_trace()
+
+    try:
+        agent_name = ctx.agent_config.agent_name
+        logger.info("Generating {} '{}' and adding it to the agent '{}'...".format(item_type, protocol_name, agent_name))
+
+        # output_path = Path(os.path.join(ctx.cwd, item_type_plural))
+        output_path = os.path.join(ctx.cwd, item_type_plural)
+        protocol_generator = ProtocolGenerator(protocol_template, output_path)
+        protocol_generator.generate()
+
+        Path(item_type_plural).mkdir(exist_ok=True)
+        #
+        # # create the connection folder
+        # dest = Path(os.path.join(item_type_plural, item_name))
+        #
+        # # copy the skill package into the agent's supported skills.
+        # src = Path(os.path.join(AEA_DIR, item_type_plural, "scaffold"))
+        # logger.debug("Copying {} modules. src={} dst={}".format(item_type, src, dest))
+        #
+        # shutil.copytree(src, dest)
+
+        # add the connection to the configurations.
+        logger.debug("Registering the {} into {}".format(item_type, DEFAULT_AEA_CONFIG_FILE))
+        existing_id_list.add(PublicId("fetchai", protocol_name, "0.1.0"))
```

Comment:
@5A11 please check for all CLI commands for the occurrence of "0.1.0". Then create a constant `DEFAULT_VERSION = 0.1.0` and use throughout instead. Magic strings/numbers are discouraged.

Reply:
Where should the constant be defined? common.py?

Reply:
I think in this case `aea.configurations.base` is most appropriate.

Reply:
Changed all occurrences except a few in docstrings.